addons/common/functions/fnc_hasItem.sqf

```diff
@@ -16,6 +17,18 @@
  * Public: Yes
  */
 
-params [["_unit", objNull, [objNull]], ["_item", "", [""]]];
+params [["_unit", objNull, [objNull]], ["_item", "", ["", [""]]], ["_checkAll", false, [false]]];
 
-_item in (_unit call EFUNC(common,uniqueItems))
+if (_item isEqualType "") then {
+    _item = [_item];
+};
+
+private _return = false;
+private _unitItems = _unit call FUNC(uniqueItems);
+
+{
+    _return = _x in _unitItems;
+    if ([_return, !_return] select _checkAll) then {break};
```

Comment:
It would be more performant to have one if condition and swap out the code passed to forEach.
Instead of evaluating every time inside the loop

Then you can also just use findIf for the !_checkAll case..
or actually can't you use findIf for both?

Reply:
is `findIf` faster than `in`?

Reply:
No.
findIf is a forEach with the exitWith/break already built in.
You're not replacing the in, you're replacing the if, then, break

Reply:
Gotcha.